scripts/pyi_generator.py

```diff
@@ -0,0 +1,144 @@
+"""The pyi generator module."""
+
+import importlib
+import inspect
+import os
+from inspect import getfullargspec
+from pathlib import Path
+from typing import Any
+
+import black
+
+from reflex.components.component import Component
+
+
+def _get_type_hint(value):
+    res = ""
+    if hasattr(value, "_name") and value._name == "Optional":
+        res = value.__args__[0].__name__
+    elif isinstance(value, str):
+        res = value
+    else:
+        res = value.__name__
+    return res
+
+
+class PyiGenerator:
+    """A .pyi file generator that will scan all defined Component in Reflex and
+    generate the approriate stub.
+    """
+
+    modules: list = []
+    root: str = ""
+    current_module: Any = {}
+
+    def _generate_imports(self, classes):
+        bases = {
+            base
+            for _, _class in classes
+            for base in _class.__bases__
+            if inspect.getmodule(base) != self.current_module
+        }
+        return [
+            "from typing import overload, Optional, Any",
+            *[f"from {base.__module__} import {base.__name__}" for base in bases],
+            "from reflex.vars import Var",
+        ]
+
+    def _generate_pyi_class(self, _class):
+        create_spec = getfullargspec(_class.create)
+        lines = [
+            "",
+            f"class {_class.__name__}({', '.join([base.__name__ for base in _class.__bases__])}):",
+        ]
+        definition = f"    @overload\n    @classmethod\n    def create(cls, *children, "
+
+        for kwarg in create_spec.kwonlyargs:
+            if kwarg in create_spec.annotations:
+                definition += (
+                    f"{kwarg}: {_get_type_hint(create_spec.annotations[kwarg])}, "
+                )
+            else:
+                definition += f"{kwarg}, "
+
+        for name, value in _class.__annotations__.items():
+            if name in create_spec.kwonlyargs:
+                continue
+            definition += f"{name}: {_get_type_hint(value)}, "
```

Comment:
the `Var` seem to be losing their arg; so `Var[bool]` just becomes `Var`.

Personally, I think these should be typed as `Var[bool] | bool` so that users don't get type checking errors when passing bare values that aren't `Var`-ified